diagnostic.js

```diff
@@ -30,25 +30,68 @@ mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
+'use strict';
+
+const models = require('app/models');
+const Movement = models.movement;
+
+const create = (name, description, startYear, endYear) => {
+  Movement.create(name, description, startYear, endYear)
+    .then(movement => {
+      return movement.save();
+    })
+    .then(movement => print(movement.json()))
+    .catch(err => console.error(err));
+};
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  Movement.find()
+    .then(movements => print(movements.json())
+    .catch(err => console.error(err));
+};
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id)
+    .then(movement => movement ? print(movements.json() : next())  // If you find an book, send the JSON,
+    .catch(err => next(err));                          // otherwise go to the next action
+};
 // Success -> If the specified Movement exists, print it as JSON;
 //              otherwise, print "Not Found" and exit.
 // Failure -> Console.error
 
-const update = (id, field, value) => {};
+const update = (id, field, value) => {
+  let movement = {_id: id};
+  Movement.findOneAndUpdate(movement, field, value)
+  .then(movement => {
+    if (!movement) {
+      return next();
+    }
+
+    return movement.update(movement, field, value)
+      .then(movement => print(movement.json()));
+  })
+  .catch(err => console.error(err));
+};
 // Success -> If the specified Movement exists, update it and print the
 //              updated Movement as JSON; otherwise, print "Not Found" and exit.
 // Failure -> Console.error
 
-const destroy = (id) => {};
+const destroy = (id) => {
+  Movement.findById(id)
+  .then( movement => {
+    if (!movement) {
+      return next();
+    }
+    movement.remove();
+    print('removed');
+  })
+  .catch(() => print('Not Found'))
+  .then(() => exit();
+};
```

Comment:
This is pretty close. You found the Movement, passed it to the promise, and if you didn't find the desired Movement, you terminated early. If it was found, you `remove`d it and printed text (or you would have, if `print` were defined).

However, as previously mentioned, `next` is specific to Express, not Promises. If you wanted to trigger a `catch`, you could have thrown an error. But in this case, all you needed to do if a record wasn't found was to print out "Not Found" and exit early by calling `return;`